bioptim/dynamics/fatigue_dynamics.py

```diff
@@ -0,0 +1,301 @@
+        F: int
+            Joint fibers recovery rate
+        R: int
+            Joint fibers relaxation rate
+        tau_max:
+            The maximum positive or negative torque
+        """
+
+        self.LR = LR
+        self.LD = LD
+        self.F = F
+        self.R = R
+        self.tau_max = tau_max
+
+
+class FatigueDynamics(OptionGeneric):
+    def __init__(self, list_index, phase):
+        super(FatigueDynamics, self).__init__(phase=phase, list_index=list_index)
+
+    def dynamics(self, dxdt, nlp, index, states, controls):
+        raise NotImplementedError("FatigueDynamics is abstract")
+
+    @property
+    def shape(self):
+        raise NotImplementedError("FatigueDynamics is abstract")
+
+
+class XiaMuscleFatigueDynamics(FatigueDynamics):
+    """
+    A placeholder for fatigue dynamics.
+    """
+
+    def __init__(
+        self,
+        LD: float,
+        LR: float,
+        F: float,
+        R: float,
+        list_index: int,
+        phase: int = 0,
+    ):
+        """
+        Parameters
+        ----------
+        LD: int
+            Joint development coefficient
+        LR: int
+            Joint relaxation coefficient
+        F: int
+            Joint fibers recovery rate
+        R: int
+            Joint fibers relaxation rate
+        list_index: int
+            The index of the structure on which the current dynamics are applied
+        phase: int
+            At which phase this objective function must be applied
+        """
+
+        super(XiaMuscleFatigueDynamics, self).__init__(list_index=list_index, phase=phase)
+        self.LR = LR
+        self.LD = LD
+        self.F = F
+        self.R = R
+
+    def apply_dynamics(self, TL, ma, mr, mf):
+        # Implementation of Xia dynamics
+        c = if_else(lt(ma, TL), if_else(gt(mr, TL - ma), self.LD * (TL - ma), self.LD * mr), self.LR * (TL - ma))
+        madot = c - self.F * ma
+        mrdot = -c + self.R * mf
+        mfdot = self.F * ma - self.R * mf + 100 * (1 - (ma + mr + mf))
+        return vertcat(madot, mrdot, mfdot)
+
+    def dynamics(self, dxdt, nlp, index, states, controls):
+        if "muscles" not in nlp.controls:
+            raise NotImplementedError("Fatigue dynamics without muscle controls is not implemented yet")
+        TL = DynamicsFunctions.get(nlp.controls["muscles"], controls)[index, :]
+        ma = DynamicsFunctions.get(nlp.states["muscles_ma"], states)[index, :]
+        mr = DynamicsFunctions.get(nlp.states["muscles_mr"], states)[index, :]
+        mf = DynamicsFunctions.get(nlp.states["muscles_mf"], states)[index, :]
+        current_dxdt = self.apply_dynamics(TL, ma, mr, mf)
+
+        dxdt[nlp.states["muscles_ma"].index[index], :] = current_dxdt[0]
+        dxdt[nlp.states["muscles_mr"].index[index], :] = current_dxdt[1]
+        dxdt[nlp.states["muscles_mf"].index[index], :] = current_dxdt[2]
+
+        return dxdt
+
+    @property
+    def shape(self):
+        return 3
```

Comment:
What is this shape for? It is never used. If you ever hard coded the number 3 (in forloop for instance) instead of using shape, please use shape.